lib/assessment.rb

```diff
@@ -5,7 +5,9 @@ class Assessment
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    long = 0
+    enum.reduce(0) { |prev, current| current > prev ? long = current : long}
+    puts long
```

Comment:
You're thinking along the right lines, but there are a couple of problems with this answer. 
1. You're comparing `current` and `prev` directly, instead of comparing their lengths.
2. This method will return `nil` instead of the longest element.
3. The `long` variable is ultimately unnecessary, since `reduce` will automatically return the last result of its block.